off_chain_data/application-go/blockParser.go

```diff
@@ -0,0 +1,397 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-gateway/pkg/identity"
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-protos-go-apiv2/ledger/rwset"
+	"github.com/hyperledger/fabric-protos-go-apiv2/ledger/rwset/kvrwset"
+	"github.com/hyperledger/fabric-protos-go-apiv2/msp"
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+type Block interface {
+	GetNumber() uint64
+	GetTransactions() []Transaction
+	ToProto() *common.Block
+}
+
+type ParsedBlock struct {
+	block           *common.Block
+	validationCodes []byte
+	transactions    []Transaction
+}
+
+func NewParsedBlock(block *common.Block) Block {
+	validationCodes := getTransactionValidationCodes(block)
+
+	return &ParsedBlock{block, validationCodes, nil}
+}
+
+func (pb *ParsedBlock) GetNumber() uint64 {
+	header := assertDefined(pb.block.GetHeader(), "missing block header")
+	return header.GetNumber()
+}
+
+// TODO: needs cache, getPayloads, parsePayload
+func (pb *ParsedBlock) GetTransactions() []Transaction {
+	return nil
+}
+
+func (pb *ParsedBlock) ToProto() *common.Block {
+	return nil
+}
+
+// Implements identity.Identity Interface
+type IdentityImpl struct {
```

Comment:
If it's really an implementation class only created by implementation of this package then I would make it private (first character lower case). Often an interface is not used and the concrete class is returned, but reusing the fabric-gateway Identity interface might make sense here.

Reply:
I will review and update the encapsulation once I refactored the code into the parser package.